src/Symfony/Bundle/FrameworkBundle/Resources/config/translation.xml

```diff
@@ -147,8 +148,8 @@
             <argument type="service" id="Psr\Container\ContainerInterface" />
         </service>
 
-        <service id="translator_listener" class="Symfony\Component\HttpKernel\EventListener\TranslatorListener">
-            <argument type="service" id="translator" />
+        <service id="locale_aware_listener" class="Symfony\Component\HttpKernel\EventListener\LocaleAwareListener">
+            <argument type="tagged" tag="locale_aware"/>
```

Comment:
using a dedicated compiler pass rather than using `tagged` could allow to unregister the listener in case it has no locale aware services at all (which becomes relevant once the service is defined even when the translator is not configured, as it allows cases where we don't have locale aware services in the core).